test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -276,3 +278,133 @@
+                                         "/ssl/ca/ca_crt.pem")
+                          :ssl-key (str bootstrap/master-conf-dir
+                                        "/ssl/ca/ca_key.pem")
+                          :ssl-ca-cert (str bootstrap/master-conf-dir
+                                            "/ssl/ca/ca_crt.pem")
+                          :as :text}
+                 _ (ssl-utils/key->pem! private-key private-key-file)
+                 _ (ssl-utils/obj->pem! csr (str bootstrap/master-conf-dir
+                                                "/ssl/ca/requests/"
+                                                subject
+                                                ".pem"))
+                 cert-status-request (fn [action]
+                                       (http-client/put
+                                         (str "https://localhost:8140/"
+                                              "puppet-ca/v1/certificate_status/"
+                                              subject)
+                                         (merge options
+                                                {:body (str "{\"desired_state\": \""
+                                                            action
+                                                            "\"}")
+                                                 :headers {"content-type"
+                                                           "application/json"}})))
+                 client-request #(http-client/get
+                                  "https://localhost:8140/status/v1/services"
+                                   (merge options
+                                          {:ssl-key (str private-key-file)
+                                           :ssl-cert (str bootstrap/master-conf-dir
+                                                          "/ssl/ca/signed/"
+                                                          subject
+                                                          ".pem")}))]
+             (testing "node certificate request can be signed successfully"
+               (let [sign-response (cert-status-request "signed")]
+                 (is (= 204 (:status sign-response)))))
+             (testing "node request before revocation is successful"
+               (let [node-response-before-revoke (client-request)]
+                 (is (= 200 (:status node-response-before-revoke)))))
+             (testing "node certificate can be successfully revoked"
+               (let [revoke-response (cert-status-request "revoked")]
+                 (is (= 204 (:status revoke-response)))))
+             (testing "node request after revocation fails"
+               (let [ssl-exception-for-request? #(try
+                                                   (client-request)
+                                                   false
+                                                   (catch SSLException e
```

Comment:
In the newer version of Jetty, the exception thrown here has apparently changed from ConnectionClosed to SSLException.

Reply:
reminds me of https://github.com/puppetlabs/puppetserver/commit/fd17c79e85a3b8584feedc75edee554a0040a724

Reply:
I think I didn't bring that commit in with this because it seemed Java 7 specific.

Reply:
Yep that was my hypothesis at the time.. maybe I was wrong

Reply:
Well I haven't seen any IOExceptions. All that SSLException, which is definitely tied to newer Jetty.